tests/lsan.supp

```diff
@@ -11,27 +11,59 @@ leak:qd_policy_open_fetch_settings
 leak:qdr_error_description
 
 # DISPATCH-1844 - shutdown leak
-leak:sys_mutex
+#leak:sys_mutex
 
 # expected, not a bug:
 #
-leak:qdr_core_subscribe
+#leak:qdr_core_subscribe
 
 # Ubuntu 16.04 (Xenial)
 #
 leak:_ctypes_alloc_format_string
 leak:__strdup
 
+# to be triaged; system_tests_http
+leak:^callback_healthz$
+leak:^callback_metrics$
+
+# to be triaged; system_tests_http1_adaptor
+leak:^pn_condition$
+leak:^pn_raw_connection$
+
 ####
-#### Miscellaneous 3rd party libraries, test code, etc:
+#### Miscellaneous 3rd party libraries:
 ####
 
-leak:*libpython*
-leak:*libwebsockets*
-leak:*python2*
+# these leaks happen even after simple Py_Initialize(); Py_Finalize();
+#  https://bugs.python.org/issue1635741
+leak:^_PyObject_Realloc
+leak:^PyObject_Malloc$
+leak:^PyThread_allocate_lock$
+
+# the PyMalloc mechanism is incompatible with Valgrind, it must be disabled or reported "leaks" must be suppressed
+#  https://pythonextensionpatterns.readthedocs.io/en/latest/debugging/debug_python.html#debug-version-of-python-memory-alloc-label
+leak:^PyMem_Malloc$
+leak:^_PyObject_GC_Resize$
+# Python uses these alloc functions if you define PYTHONDEVMODE=1
+leak:^_PyMem_DebugRawAlloc$
+leak:^_PyMem_DebugRawRealloc$
+# All the rest
+leak:^list_append$
 
-# We should be able to uncomment these once all known dispatch leaks have been fixed
-leak:*libqpid-proton*
+# Wholesale library suppressions, avoid doing this if at all possible
+#  one reason we may not be able to avoid doing this is compatibility with multiple versions of libs on many operating systems
+#leak:/libpython2.*.so
+#leak:/libpython3.*.so
+#leak:/libwebsockets.so
+#leak:libqpid-proton
 
-# Ignore test code
-leak:run_unit_tests.c
+# We might be able to remove these once all known dispatch leaks have been fixed
+# Suppressions taken from Proton's lsan.supp
+#  this appears in system_tests_open_properties:
+leak:^pni_data_grow$
+leak:^pn_buffer_ensure$
+#  this appears in system_tests_http1_adaptor
+leak:^pn_string_grow$
+leak:^pn_object_new$
+leak:^pn_list$
+leak:^pni_record_create$
```

Comment:
these are known leaks in Proton, taken straight out from its lsan.supp. I think it is reasonable to carry this here for now, assuming the proton leaks get fixed in the foreseeable future